src/services/actions.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * name : services/actions.js
+ * author : Priyanka Pradeep
+ * Date : 13-Aug-2024
+ * Description : Activities Service
+ */
+const httpStatusCode = require('@generics/http-status')
+const actionQueries = require('@database/queries/actions')
+const responses = require('@helpers/responses')
+const { UniqueConstraintError } = require('sequelize')
+const common = require('@constants/common')
+const { Op } = require('sequelize')
+module.exports = class ActionsHelper {
+	/**
+	 * Create Action
+	 * @method
+	 * @name create
+	 * @param {String} bodyData - action creation data
+	 * @returns {JSON} - action creation response
+	 */
+	static async create(bodyData) {
+		try {
+			const action = await actionQueries.findOne({ code: bodyData.code })
+			if (action?.id) {
```

Comment:
@priyanka-TL why do we have to do this check? It will automatically fail because of the DB index right?